be/src/exec/vectorized/connector_scan_node.cpp

```diff
@@ -350,6 +357,46 @@ bool ConnectorScanNode::_submit_scanner(ConnectorScanner* scanner, bool blockabl
     return false;
 }
 
+bool ConnectorScanNode::_submit_streaming_load_scanner(ConnectorScanner* scanner, bool blockable) {
+    ThreadPool* thread_pool = _runtime_state->exec_env()->streaming_load_thread_pool();
+    _running_threads.fetch_add(1, std::memory_order_release);
+    // Assume the thread pool is large enough, so there is no need to set the priority
+    Status status = thread_pool->submit_func([this, scanner] { _scanner_thread(scanner); });
+    if (status.ok()) {
+        return true;
+    }
+
+    // Thread pool for streaming load is assumed to be infinite, and submit should not fail
+    // because of the pool is full. The possible failure reason is that the thread pool is
+    // shutdown. But we still implement a block logic like PriorityPool::offer used in
+    // _submit_scanner to handle unexpected cases.
+    if (blockable) {
+        Status block_status;
+        while (thread_pool->is_pool_status_ok()) {
+            sleep(1);
```

Comment:
why?

Reply:
Here I want to implement a block logic similar to PriorityThreadPool::offer, but in general this logic will not be reached if the status of thread pool is ok because the pool is large enough. It's just a safe guard.